activerecord/lib/active_record/connection_adapters/statement_pool.rb

```diff
@@ -3,9 +3,11 @@ module ConnectionAdapters
     class StatementPool
       include Enumerable
 
-      def initialize(max = 1000)
+      DEFAULT_STATEMENT_LIMIT = 1000
+
+      def initialize(statement_limit = DEFAULT_STATEMENT_LIMIT)
         @cache = Hash.new { |h,pid| h[pid] = {} }
-        @max = max
+        @statement_limit = statement_limit
```

Comment:
Hmm, I had tried previous version, hence was not passing nil here and instead DEFAULT_STATEMENT_LIMIT. 
Let me try it out.